apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerTestUtilsV2.java

```diff
@@ -1011,4 +1017,316 @@ static void assertPeriodEquals(
+        .setSystem(system)
+        .setValue(value);
+  }
+
+  /**
+   * Finds an {@link Extension} in a list based on the Extension URL
+   *
+   * @param url
+   * @param extensions
+   */
+  static Extension findExtensionByUrl(String url, List<Extension> extensions) {
+    Optional<Extension> ex = extensions.stream().filter(e -> url.equals(e.getUrl())).findFirst();
+
+    Assert.assertTrue(ex.isPresent());
+
+    return ex.get();
+  }
+
+  /**
+   * Finds a specific {@link Coding} in a list given the system
+   *
+   * @param system
+   * @param codings
+   */
+  static Coding findCodingBySystem(String system, List<Coding> codings) {
+    Optional<Coding> coding =
+        codings.stream().filter(c -> system.equals(c.getSystem())).findFirst();
+
+    Assert.assertTrue(coding.isPresent());
+
+    return coding.get();
+  }
+
+  /**
+   * Finds a Care Team member by Sequence value
+   *
+   * @param seq
+   * @param team
+   */
+  static CareTeamComponent findCareTeamBySequence(int seq, List<CareTeamComponent> team) {
+    Optional<CareTeamComponent> ctc = team.stream().filter(c -> c.getSequence() == seq).findFirst();
+
+    Assert.assertTrue(ctc.isPresent());
+
+    return ctc.get();
+  }
+
+  /**
+   * Helper that creates a {@link CareTeamComponent} to be used in unit tests
+   *
+   * @param sequence The sequence to set
+   * @param npi The NPI for the member
+   * @param system System defining the type of member
+   * @param code Code defining the type of member
+   * @param display Display for the type of member
+   * @return {@link CareTeamComponent}
+   */
+  static CareTeamComponent createNpiCareTeamMember(
+      int sequence, String npi, String system, String code, String display) {
+    return new CareTeamComponent()
+        .setSequence(sequence)
+        .setProvider(
+            new Reference()
+                .setIdentifier(
+                    createIdentifier(
+                        null,
+                        npi,
+                        "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBIdentifierType",
+                        "npi",
+                        "National Provider Identifier")))
+        .setRole(new CodeableConcept().setCoding(Arrays.asList(new Coding(system, code, display))));
+  }
+
+  /**
+   * Finds a {@link SupportingInformationComponent} based on the value of a Code of the Category
+   *
+   * @param code
+   * @param components
+   * @return
+   */
+  static SupportingInformationComponent findSupportingInfoByCode(
+      String code, List<SupportingInformationComponent> components) {
+    Optional<SupportingInformationComponent> si =
+        components.stream()
+            .filter(
+                cmp ->
+                    cmp.getCategory().getCoding().stream()
+                            .filter(c -> code.equals(c.getCode()))
+                            .count()
+                        > 0)
+            .findFirst();
+
+    Assert.assertTrue(si.isPresent());
+
```

Comment:
I had a comment that I nuked in the Transformer tests r.e. Asserting we have an object prior to constructing the element for the deep compare. But I should have checked here first...

Reply:
:+1:

Yeah this seemed like a more appropriate place to check that we got a value back than in every individual test.  I'm not sure I really liked this, because as you noticed it is a bit opaque .. but it does save some copy pasting.